launcher/minecraft/mod/Mod.cpp

```diff
@@ -181,106 +134,86 @@ auto Mod::name() const -> QString
     if (!d_name.isEmpty())
         return d_name;
 
-    if (metadata())
-        return metadata()->name;
-
-    return m_name;
+    return Resource::name();
 }
 
 auto Mod::version() const -> QString
 {
     return details().version;
 }
 
-auto Mod::homeurl() const -> QString
+auto Mod::homepage() const -> QString
 {
-    return details().homeurl;
+    QString metaUrl = Resource::homepage();
+
+    if (metaUrl.isEmpty())
+        return details().homeurl;
+    else
+        return metaUrl;
 }
 
-auto Mod::metaurl() const -> QString
+auto Mod::loaders() const -> QString
 {
-    if (metadata() == nullptr)
-        return homeurl();
-    return ModPlatform::getMetaURL(metadata()->provider, metadata()->project_id);
+    if (metadata()) {
+        QStringList loaders;
+        auto modLoaders = metadata()->loaders;
+        for (auto loader : { ModPlatform::NeoForge, ModPlatform::Forge, ModPlatform::Cauldron, ModPlatform::LiteLoader, ModPlatform::Fabric,
+                             ModPlatform::Quilt }) {
```

Comment:
This should probably be a static list or returned by a static function somewhere, so it's easy to find and we only ever need to update it in once place.

Reply:
Out of scope